homeworks/05-redux-basic/src/App.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { createStore, Action, Reducer } from 'redux'
+import { devToolsEnhancer } from 'redux-devtools-extension'
+import './App.css'
+
+enum AccountActionType {
+  UPDATE_BALANCE = "UPDATE_BALANCE",
+  CREDIT = "CREDIT",
+  DEBIT = "DEBIT",
+  GET_BALANCE_WITHOUT_TAX = "GET_BALANCE_WITHOUT_TAX",
+  GET_BALANCE_WITH_TAX = "GET_BALANCE_WITH_TAX"
+}
+
+type AccountActionTypeKey = keyof typeof AccountActionType
+
+interface AccountAction extends Action<AccountActionTypeKey> {
+  payload?: number
+}
+
+type AccountState = {
+  balance: number,
+  history: AccountAction[]
+}
+
+const reducer: Reducer<AccountState, AccountAction> = (state = { balance: 0, history: [] }, action) => {
+  let balance = state.balance
+  switch (action.type) {
+    case AccountActionType.UPDATE_BALANCE:
+    case AccountActionType.CREDIT:
+      if (action.payload) {
+        balance += action.payload
+      }
+      break
+    case AccountActionType.DEBIT:
+      if (action.payload) {
```

Comment:
It would be nice to avoid duplication of code. I'd say there are several ways how to manage it. For instance, create utils method that is able to both verify it and set default value. I mean to say that liner structure of code is better for further extending.

Reply:
Is there any good way to do it, given that some actions have payload and others - don't have? Only 1 line is repeated here, extracting it to function doesn't make sense.